### PR TITLE
Add float16 vector

### PR DESCRIPTION
**Pull request by Writer-X**
#22837

### CODE REVIEW

File: internal/core/src/query/SearchBruteForce.cpp [hunk 1]
```diff
@@ -50,6 +52,13 @@ BruteForceSearch(const dataset::SearchDataset& dataset,
         auto dim = dataset.dim;
         auto topk = dataset.topk;
 
+        bool isFp16 = false;
+        if (data_type == DataType::VECTOR_FLOAT16)
+        {
+            std::cout << "BruteForceSearch: data_type is VECTOR_FLOAT16" << std::endl;
+            isFp16 = true;
+        }
+
```
Comment (jiaoew1991) [inline internal/core/src/query/SearchBruteForce.cpp:61]: What is this code trying to do?

**Comment by jiaoew1991** [comment]
overall LGTM, the UT was panic by segv, it seems have some unsafe memory usage.

File: scripts/download_milvus_proto.sh [hunk 2]
```diff
@@ -7,19 +7,19 @@ API_VERSION=$(go list -m github.com/milvus-io/milvus-proto/go-api/v2 | awk -F' '
 if [ ! -d "$THIRD_PARTY_DIR/milvus-proto" ]; then
   mkdir -p $THIRD_PARTY_DIR
   pushd $THIRD_PARTY_DIR
-  git clone https://github.com/milvus-io/milvus-proto.git
+  git clone https://github.com/Writer-X/milvus-proto.git
```
Comment (yah01) [inline scripts/download_milvus_proto.sh:10]: Need to restore this file before merge it

**Review by yah01 — CHANGES_REQUESTED** [review]

File: internal/core/src/query/SearchBruteForce.cpp (hunk 1 above)
Comment (Writer-X) [reply internal/core/src/query/SearchBruteForce.cpp:61]: It's an argument to a function in knowhere, but I haven't finished the knowhere part yet. So it's useless right now,

**Comment by Writer-X** [comment]
> overall LGTM, the UT was panic by segv, it seems have some unsafe memory usage.

I did not report any errors when running in the local environment, but the results were not quite correct. 
The problem may be in the knowhere section. For FP16-type queries, I'll fix the memory problem after I finish the knowhere section.

File: internal/core/src/query/SearchBruteForce.cpp [hunk 3]
```diff
@@ -50,6 +52,13 @@ BruteForceSearch(const dataset::SearchDataset& dataset,
         auto dim = dataset.dim;
         auto topk = dataset.topk;
 
+        bool isFp16 = false;
+        if (data_type == DataType::VECTOR_FLOAT16)
+        {
+            std::cout << "BruteForceSearch: data_type is VECTOR_FLOAT16" << std::endl;
+            isFp16 = true;
```
[the remark this answers is not in the document]
Comment (Writer-X) [reply internal/core/src/query/SearchBruteForce.cpp:59]: Thank you， I  will fix it before the next submission.

**Pull request closed without merging**